File: javascript/BinaryTree.js
class Node {
  constructor(data) {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}

class BinaryTree {
  constructor() {
    this.root = null;
  }
  insert(data) {
    this.root = this._insertRecursive(this.root, data);
  }
  _insertRecursive(root, data) {
    if(root === null) {
      root = new Node(data);
      return root;
    } else if(data < root.data) {
      root.left = this._insertRecursive(root.left, data);
    } else if(data > root.data) {
      root.right = this._insertRecursive(root.right, data);
    }
    return root;
  }
  inorderTraversal() {
    this._inorderRecursive(this.root);
  }
  _inorderRecursive(root) {
    if(root !== null) {
      this._inorderRecursive(root.left);
      console.log(root.data);
      this._inorderRecursive(root.right);
    }
  }
}
// example usage
const tree = new BinaryTree();
tree.insert(50);
tree.insert(30);
tree.insert(70);
tree.insert(20);
tree.insert(40);
tree.insert(60);
treen.insert(80);

tree.inorderTraversal();       //output; 20 30 40 50 60 70 80

File: python/Binary_Tree.py
class Node:
  def __init__(self, data):
    self.data = data
    self.left = None
    self.right = None

class BinaryTree:
  def __init__(self):
    self.root = None

  def insert(self, data):
    self.root = self._insert_recursive(self.root, data)

  def _insert_recursive(self, root, data):
    if root is None:
      return Node(data)
    else:
      if data < root.data:
        root.left = self._insert_recursive(root.left, data)
      else:
        root.right = self._insert_recursive(root.right, data)
    return root

  def inorder_traversal(self):
    self._inorder_recursive(self.root)

  def _inorder_recursive(self, root):
    if root:
      self._inorder_recursive(root.left)
      print(root.data, end= " ")
      self._inorder_recursive(root.right)


# example usage
tree = BinaryTree()
tree.insert(50)
tree.insert(30)
tree.insert(70)
tree.insert(20)
tree.insert(40)
tree.insert(60)
tree.insert(80)

tree.inorder_traveral()    #output; 20 30 40 50 60 70 80
